src/hidra/receiver/datareceiver.py

```diff
@@ -259,6 +260,97 @@ def stop(self):
         self.run_loop = False
 
 
+def run_plugin_thread(
+        plugin_name, plugin_config, target_dir, data_queue, log, event):
+    # Execute all plugin code on a separate thread to protect main thread from
```

Comment:
it might help to give a parameter list (including expected type) of the arguments here

Reply:
Added docstring